configurator/templates/AppServer/conf/carbon.jinja

```diff
@@ -0,0 +1,616 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<!--
+    This is the main server configuration file
+
+    ${carbon.home} represents the carbon.home system property.
+    Other system properties can be specified in a similar manner.
+-->
+<Server xmlns="http://wso2.org/projects/carbon/carbon.xml">
+
+    <!--
+       Product Name
+    -->
+    <Name>Application Server</Name>
+
+    <!--
+       machine readable unique key to identify each product
+    -->
+    <ServerKey>AS</ServerKey>
+
+    <!--
+       Product Version
+    -->
+    <Version>5.2.1</Version>
+
+    <!--
+       Host name or IP address of the machine hosting this server
+       e.g. www.wso2.org, 192.168.1.10
+       This is will become part of the End Point Reference of the
+       services deployed on this server instance.
+    -->
+    {% if stratos_instance_data_worker_host_name is defined and subDomain=="mgt" : %}
+        <HostName>{{stratos_instance_data_worker_host_name}}</HostName>
+    {% endif %}
+
+
+    <!--
+    Host name to be used for the Carbon management console
+    -->
+    {% if stratos_instance_data_mgt_host_name is defined %}
```

Comment:
This might indicate that there are certain sections that should be indicated only if a condition is true only. If we use a placeholder instead, we would be passing XML tags as well here.